packages/mds-utils/utils.ts

```diff
@@ -507,14 +497,39 @@ function getPolygon(geographies: Geography[], geography: string): Geometry | Fea
   return res.geography_json
 }
 
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+function areThereCommonElements(arr1: any[], arr2: any[]) {
+  const set = new Set([...arr1, ...arr2])
+  return set.size !== arr1.length + arr2.length
+}
+
 function isInStatesOrEvents(rule: Rule, event: VehicleEvent): boolean {
-  const status = rule.states ? rule.states[EVENT_STATES_MAP[event.event_type] as VEHICLE_STATE] : null
-  return status !== null
-    ? rule.states !== null &&
-        Object.keys(rule.states).includes(EVENT_STATES_MAP[event.event_type]) &&
-        status !== undefined &&
-        (status.length === 0 || (status as string[]).includes(event.event_type))
-    : true
+  const { states } = rule
+  // If no states are specified, then the rule applies to all VehicleStates.
+  if (states === null || states === undefined) {
+    return true
+  }
+
+  // States that it is possible to transition into with event.event_type
+  const possibleStates: VEHICLE_STATE[] = event.event_types.reduce((acc: VEHICLE_STATE[], event_type) => {
+    return acc.concat(EVENT_STATES_MAP[event_type])
+  }, [])
```

Comment:
Also nit, this method doesn't need to be wrapped in curly braces, could just be a oneliner without a `return` statement